File: LAB7/zad2.cpp
#include <iostream>  // poprawnie; ale, dlaczego rozwiązał Pan tylko dwa zadania z pięciu?

using namespace std;

class Discount {
public:
    virtual double calculateDiscount(double price) const = 0;
};


class RegularDiscount : public Discount {
public:
    double calculateDiscount(double price) const override{
        return price * 0.9;
    }
};


class SeasonalDiscount : public Discount {
public:
    double calculateDiscount(double price) const override{
        return price * 0.8;
    }
};

int main() {
    double price;
    
    cout << "Podaj cene: ";
    cin >> price;

   RegularDiscount regularDiscount;
   SeasonalDiscount seasonalDiscount;

   cout << "Cena z regularnym rabatem: " << regularDiscount.calculateDiscount(price) << endl;
   cout << "Cena z sezonowym rabatem: " << seasonalDiscount.calculateDiscount(price) << endl;

}

File: LAB7_JAVA/zad2.java
import java.util.Scanner; // poprawnie

interface Discount {
   double calculate(double price); 
}

class RegularDiscount implements Discount {
     @Override
     public double calculate(double price) {
         return 0.9*price;
     }
}


class SeasonalDiscount implements Discount {
    @Override
    public double calculate(double price) {
        return 0.8*price;
    }
}

public class zad2 {
    public static void main(String[] args) {
        RegularDiscount regularDiscount = new RegularDiscount();
        SeasonalDiscount seasonalDiscount = new SeasonalDiscount();


        Scanner scanner = new Scanner(System.in);
        System.out.println("Podaj cene:");
        double price = scanner.nextDouble();

        System.out.println("Cena po zniżce regularnej:");
        System.out.println(regularDiscount.calculate(price));

        System.out.println("Cena po zniżce sezonowej:");
        System.out.println(seasonalDiscount.calculate(price));

    }
}
